spec/helpers.lua

```diff
@@ -1163,8 +1173,13 @@ return {
   end,
   stop_kong = function(prefix, preserve_prefix, preserve_tables)
     prefix = prefix or conf.prefix
+
+    local running_conf = get_running_conf(prefix)
+    if not running_conf then return end
+
     local ok, err = kong_exec("stop --prefix " .. prefix)
-    wait_pid(conf.nginx_pid, nil)
+
+    wait_pid(running_conf.nginx_pid)
```

Comment:
This is a great fix. Maybe this warrants `--wait` flag in the CLI to allow for other users to not have to rely on custom logic when scripting `kong stop`? Maybe this should even be the default behaviour of `kong quit` (since it is the "graceful" stop).